src/index.js

```diff
@@ -2,24 +2,29 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  
+  let total = price.innerHTML * quantity.value;
 
-  //... your code goes here
+  let subtotal = product.querySelector('.subtotal span');
+
+  subtotal.innerHTML = total;
+  return total;
+  
 }
 
+
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+  let total = 0;
+  const products = document.getElementsByClassName("product");
 
-  // ITERATION 3
-  //... your code goes here
-}
+  for (let i = 0; i < products.length; i++) {
+updateSubtotal(products[i]);
```

Comment:
In this iteration you forgot to update the Total value, to do that you need to declare a variable where you will add the prices from all the products, and then you need to target the HTML tag for the total value and update it's value:

```
function calculateAll() {
  const products = document.getElementsByClassName("product");

  let sum = 0;

  for (let i = 0; i < products.length; i++) {
    updateSubtotal(products[i]);
    sum += updateSubtotal(products[i]);
  }

  let totalValue = document.querySelector("#total-value span");
  totalValue.innerHTML = sum;
}
```